### ISSUE TITLE
Embedded bplist module is failing to preserve bool type; should switch to standard library plistlib with FMT_BINARY

### ISSUE THREAD

**Issue by TheGS** [xa4a/bpylist2#5]
The embedded `bplist` module is failing to preserve the type of boolean values in generated plists, and should now be replaced with using the standard library `plistlib` module (and the `FMT_BINARY` format flag).

On Python 3, the standard library `plistlib` module supports parsing and generating binary plist format, so there is no need to use the embedded C-based bplist module anymore.

A comparison of results I got from using `bplist` ...

```
$ python
Python 3.7.3 (default, Nov 15 2019, 05:07:46) 
[Clang 11.0.3 (clang-1103.0.23.96)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from bpylist import bplist
>>> bplist.parse(bplist.generate(False))
0
>>> type(bplist.parse(bplist.generate(False)))
<class 'int'>
>>> bplist.parse(bplist.generate(True))
1
>>> type(bplist.parse(bplist.generate(True)))
<class 'int'>
```

against using `plistlib` on Python 3 ...

```
$ python
Python 3.7.3 (default, Nov 15 2019, 05:07:46) 
[Clang 11.0.3 (clang-1103.0.23.96)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import plistlib
>>> plistlib.loads(plistlib.dumps(False, fmt=plistlib.FMT_BINARY))
False
>>> type(plistlib.loads(plistlib.dumps(False, fmt=plistlib.FMT_BINARY)))
<class 'bool'>
>>> plistlib.loads(plistlib.dumps(True, fmt=plistlib.FMT_BINARY))
True
>>> type(plistlib.loads(plistlib.dumps(True, fmt=plistlib.FMT_BINARY)))
<class 'bool'>
```

**Comment by TheGS** [comment]
FWIW, my employer makes it difficult for me to contribute code/IP directly to open source projects without jumping through hoops so I can't just supply a PR, but I don't expect a fix to be too difficult.

**Comment by parabolala** [comment]
I looked into this briefly and the initial findings: 
* `plistlib` implementation doesn't support `uid` tokens until Python 3.8 (unless there's a backport as a library that I didn't find).
* Current archiver class relies heavily on `uid()`'s and their support in the underlying plist serializer. 

I'll look into swapping the native `bplist` with `plistlib` assuming 3.8 and will push a new version if I succeed. Hope the py3.8 assumption works for you.
